Description: Allow XCAS_ROOT override, so doc files can be found during the build
 By default the documentation build looks in /usr/share, leading to non-fatal
 errors like "// Unable to find keyword file doc/en/keywords".
Author: Ximin Luo <infinity0@debian.org>
Forwarded: not-needed

--- a/src/global.cc
+++ b/src/global.cc
@@ -2567,6 +2567,10 @@
     if (!access((xcasroot()+"aide_cas").c_str(),R_OK)){
       return xcasroot();
     }
+    if (getenv("XCAS_ROOT")){
+      string s=getenv("XCAS_ROOT");
+      return s;
+    }
     if (xcasroot().size()>4 && xcasroot().substr(xcasroot().size()-4,4)=="bin/"){
       string s(xcasroot().substr(0,xcasroot().size()-4));
       s+="share/giac/";
@@ -2577,10 +2581,6 @@
 #ifdef __APPLE__
     if (!access("/Applications/usr/share/giac/",R_OK))
       return "/Applications/usr/share/giac/";
-    if (getenv("XCAS_ROOT")){
-      string s=getenv("XCAS_ROOT");
-      return s;
-    }
     return "/Applications/usr/share/giac/";
 #endif
 #ifdef WIN32
